packages/file_selector/file_selector_platform_interface/lib/src/types/x_type_group/x_type_group.dart

```diff
@@ -0,0 +1,53 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// TODO: should we be using this package or just extracting the important conversion data from it?
+import 'package:mime_type/mime_type.dart';
+
+/// A set of allowed XTypes
+class XTypeGroup {
+  /// Creates a new group with the given label and file extensions.
+  XTypeGroup({
+    this.label,
+    this.extensions,
+    this.mimeTypes,
+    this.macUTIs,
+    this.webWildCards,
+  }) {
+    if (
+      this.extensions == null &&
+      this.mimeTypes == null &&
+      this.macUTIs == null &&
+      this.webWildCards == null
+    ) {
+      throw ArgumentError("At least one type must be provided for an XTypeGroup.");
+    }
```

Comment:
You may be able to do this through an assertion in the initialization list of a constructor without body:

```dart
XTypeGroup({
  this.label,
  this.extensions,
  ...
}) : assert(!(extensions == null && mimeTypes == null && ...), 'At least one type must be provided for an XTypeGroup');
```

See more: https://dart.dev/guides/language/language-tour#initializer-list